src/ostree/ot-builtin-summary.c

```diff
@@ -40,6 +40,63 @@ static GOptionEntry options[] = {
   { NULL }
 };
 
+/* Take arguments of the form [key=value] or [key, value] out of @argv and put
+ * them into an a{sv} variant. The value arguments must be parsable using
+ * g_variant_parse(). */
+static GVariant *
+argv_to_additional_metadata (int      *argc,
+                             char   ***argv,
+                             GError  **error)
+{
+  g_autoptr(GVariantBuilder) builder = NULL;
+  int i;
+
+  builder = g_variant_builder_new (G_VARIANT_TYPE_VARDICT);
+
+  for (i = 1; i < *argc; i++)
+    {
+      const gchar *equals = strchr ((*argv)[i], '=');
+      g_autofree gchar *key = NULL;
+      const gchar *value_str;
+      g_autoptr(GVariant) value = NULL;
+
+      if (equals != NULL)
+        {
+          key = g_strndup ((*argv)[i], equals - (*argv)[i]);
+          value_str = equals + 1;
+        }
+      else
+        {
+          key = g_strdup ((*argv)[i]);
+
+          if (i + 1 >= *argc)
+            {
+              g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "Invalid additional metadata list: not an even number of arguments");
+              return NULL;
+            }
+
+          value_str = (*argv)[i + 1];
+          i++;
+        }
+
+      value = g_variant_parse (NULL, value_str, NULL, NULL, error);
+      if (value == NULL)
+        {
+          g_prefix_error (error, "Error parsing variant ‘%s’: ", value_str);
+          return NULL;
+        }
+
+      g_variant_builder_add (builder, "{sv}", key, value);
+    }
+
+  /* Update the argument list. */
+  for (; i > 1; i--)
+    (*argv)[i] = NULL;
+
+  return g_variant_builder_end (builder);
```

Comment:
Generally we `ref_sink()` variants; with autocleanups it's less useful to have things floating, and can be actually a danger :skull_and_crossbones:  if one was using an autocleanup and happened to later pass a floating variant to a function which sinks it.